src/endpoints/process-nfts/process.nfts.service.ts

```diff
@@ -44,4 +90,48 @@ export class ProcessNftsService {
 
     return await this.nftWorkerService.addProcessNftQueueJob(nft, settings);
   }
+
+  private async checkCollectionOwner(address: string, collection?: string, identifier?: string): Promise<void> {
+    if (identifier) {
+      const nft = await this.nftService.getSingleNft(identifier);
+      if (!nft) {
+        throw new HttpException('Provide a valid identifier or a collection to generate thumbnails for', HttpStatus.BAD_REQUEST);
```

Comment:
at this level you should throw normal errors, since theoretically the service layer is not dependent on http calls.
You can try/catch it in the controller and re-package the error there as BadRequest
Also, you can throw new BadRequestException which does the same but is a bit better integrated